src/components/DecisioningEngine/createEvaluableRulesetPayload.js

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import RulesEngine from "@adobe/aep-rules-engine";
+import { JSON_RULESET_ITEM } from "../Personalization/constants/schema";
+import flattenArray from "../../utils/flattenArray";
+
+export default payload => {
+  const items = [];
+
+  const addItem = item => {
+    const { data = {} } = item;
+    const { content } = data;
+
+    if (!content) {
+      return;
+    }
+
+    items.push(RulesEngine(JSON.parse(content)));
+  };
+
+  const evaluate = context => {
+    return {
+      ...payload,
+      items: flattenArray(items.map(item => item.execute(context))).map(
```

Comment:
Could you get away with just using flatMap here? We already have defined that.

Reply:
i tried but I couldn't get it to work how i liked, and it was ugly nested flatMaps